drafts/signals-everywhere.md

```diff
@@ -0,0 +1,167 @@
+
+```js
+function App() {
+    const [firstName, setFirstName] = createSignal('Bob');
+    const [lastName, setLastName] = createSignal('Smith');
+    const fullName = createMemo(() => `${firstName()} ${lastName()}`);
+}
+```
+
+If we only look at the surface, it seems like the progress of web development in over a decade has been insignificant.
+Which of course is not true.
+But using fine-grained reactivity is certainly a comeback – and it's a strong one.
+Aside from React, all major frameworks have introduced Signals or a similar concept in the last few years.
+
+As Ryan Carniato described in his article [The Quest for ReactiveScript](https://dev.to/this-is-learning/the-quest-for-reactivescript-3ka3) in 2021, there are commonly three types of reactivity:
+
+- Reactive State: Signal, Observable, Ref
+- Derived Values: Memo, Computed
+- Side Effects: Effect, Watch, Reaction, Autorun
+
+The following table is an overview of how the implementation of these types is called in their respective framework.
+
+| Framework | Signals since | Reactive State | Derived Values | Side Effect |
+|-|-|-|-|-|
+| Knockout | 2010 | observable | pureComputed | computed |
+| Svelte | 2016 | writable/readable | derived | – (reactive statements) |
+| Solid | 2018 | createSignal | createMemo | createEffect |
+| Vue | 2020 | ref/reactive | computed | watch/watchEffect |
+| Preact | 2022 | signal | computed | effect |
+| Qwik | 2022 | useSignal/useStore | useComputed$/useResource$ | useTask$ |
+| Angular | 2023 | signal | computed | effect |
+| Svelte 5 | 2024 | $state | $derived | $effect |
+
+Solid made Signals the talk of the town and has done the web community a big favour.
+For the first time, we have a robust, coherent, somewhat standardised approach for reactive user interfaces across multiple frameworks.
+The web is converging a little.
+**Signals have become transferrable knowledge**, making it easier and faster to learn a different technology.
+Which brings us – finally – to the question: what are they?
+
+## Signals explained
+
+The core idea is unchanged from Knockout's definition of observables: "special objects, which notify subscribers and can detect dependencies".
+A Signal gives you a getter and a setter, while other state solutions (like React's `useState`) give you the value and a setter.
+Having a getter means you can **pass the reference** to the Signal, preserving its reactivity.
+Calling the getter is secretly subscribing to the value ergonomically, letting the library deal with managing the subscriptions.
+
+Whether you get two separate functions like in Solid or a single object with a `.value` property (Preact, Qwik, …) doesn't matter.
+Those are implementation details and don't change what's happening behind the scenes.
+
+These source Signals can be used to construct new ones, building a reactive chain of Signals.
+Instead of applying the same operations on a "raw" Signal in multiple places, we can offer the result itself as a reactive value.
```

Comment:
is there a way to maybe show some code with the Bob Smith example and createMemo?

Reply:
I don't think it would add anything useful here. It would be pretty much the same code again, bloating the article unnecessarily. I thought about adding a reference like "see the Bob Smith example from before" but I don't want readers to have to jump around either.